File: android/app/src/main/java/com/yihengquan/sendtext/WebServer.java
package com.yihengquan.sendtext;

import java.io.UnsupportedEncodingException;
import java.nio.ByteBuffer;
import java.nio.charset.StandardCharsets;

import static java.nio.charset.StandardCharsets.*;
import fi.iki.elonen.NanoHTTPD;

public class WebServer extends NanoHTTPD {

    private String message;

    public WebServer(int port, String message) {
        super(port);
        this.message = message;
    }

    public void setMessage(String msg) {
        this.message = msg;
    }

    @Override
    public Response serve(IHTTPSession session) {
        String encode = "";
        try {
            // Working utf-8 encoding
            byte[] kanji = message.getBytes("UTF-8");
            for (byte b : kanji) {
                encode += String.format("\\x%2x", b);
            }
        } catch (UnsupportedEncodingException e) {
            e.printStackTrace();
        }
        return newFixedLengthResponse(encode);
    }

}

File: android/app/src/main/java/com/yihengquan/sendtext/WebServer.kt
package com.yihengquan.sendtext

import fi.iki.elonen.NanoHTTPD
import java.io.UnsupportedEncodingException

class WebServer(port: Int, private var message: String) : NanoHTTPD(port) {
    fun setMessage(msg: String) {
        message = msg
    }

    override fun serve(session: IHTTPSession): Response {
        var encode = ""
        try {
            // Working utf-8 encoding
            val kanji = message.toByteArray(charset("UTF-8"))
            for (b in kanji) {
                encode += String.format("\\x%2x", b)
            }
        } catch (e: UnsupportedEncodingException) {
            e.printStackTrace()
        }
        return newFixedLengthResponse(encode)
    }

}